homeassistant/components/lg_thinq/sensor.py

```diff
@@ -0,0 +1,590 @@
+"""Support for sensor entities."""
+
+from __future__ import annotations
+
+import logging
+
+from thinqconnect import DeviceType
+from thinqconnect.devices.const import Property as ThinQProperty
+from thinqconnect.integration import ThinQPropertyEx, TimerProperty
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import (
+    CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+    PERCENTAGE,
+    UnitOfTemperature,
+    UnitOfTime,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ThinqConfigEntry
+from .coordinator import DeviceDataUpdateCoordinator
+from .entity import ThinQEntity
+
+COMMON_SENSOR_DESC: dict[ThinQProperty, SensorEntityDescription] = {
+    ThinQPropertyEx.ERROR: SensorEntityDescription(
+        key=ThinQPropertyEx.ERROR,
+        translation_key=ThinQPropertyEx.ERROR,
+    ),
+    ThinQPropertyEx.NOTIFICATION: SensorEntityDescription(
+        key=ThinQPropertyEx.NOTIFICATION,
+        translation_key=ThinQPropertyEx.NOTIFICATION,
+    ),
+}
+AIR_QUALITY_SENSOR_DESC: dict[ThinQProperty, SensorEntityDescription] = {
+    ThinQProperty.PM1: SensorEntityDescription(
+        key=ThinQProperty.PM1,
+        device_class=SensorDeviceClass.PM1,
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        state_class=SensorStateClass.MEASUREMENT,
+        translation_key=ThinQProperty.PM1,
+    ),
+    ThinQProperty.PM2: SensorEntityDescription(
+        key=ThinQProperty.PM2,
+        device_class=SensorDeviceClass.PM25,
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        state_class=SensorStateClass.MEASUREMENT,
+        translation_key=ThinQProperty.PM2,
+    ),
+    ThinQProperty.PM10: SensorEntityDescription(
+        key=ThinQProperty.PM10,
+        device_class=SensorDeviceClass.PM10,
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        state_class=SensorStateClass.MEASUREMENT,
+        translation_key=ThinQProperty.PM10,
+    ),
+    ThinQProperty.HUMIDITY: SensorEntityDescription(
+        key=ThinQProperty.HUMIDITY,
+        device_class=SensorDeviceClass.HUMIDITY,
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        translation_key=ThinQProperty.HUMIDITY,
+    ),
+    ThinQProperty.ODOR: SensorEntityDescription(
+        key=ThinQProperty.ODOR,
+        translation_key=ThinQProperty.ODOR,
+    ),
```

Comment:
ODR is not enum but number.
Because we get number from the cloud, we don't have any enum information.

Reply:
Are the numbers 0-4 set in stone? I would recommend transforming that using a dict
```py
{
    0: "weak"
}
```
And then we can provide a list of options to the entity description

Reply:
We don't use that way in other environments.
And I am not sure it has small range of numbers.

Reply:
This could be a dictionary in the HA code, but this way we transform it to something user readable, because the untranslated values are still used in templating, so I want to avoid using numbers

Reply:
Odor property has not yet supported enum llist. I'll remove it's state translation in strings.json.
        "state": {
          "1": "Weak",
          "2": "Normal",
          "3": "Strong",
          "4": "Very strong"
        }

Reply:
We do want these translations, but we should transform them to snake case strings

Reply:
I have question. Does this mean can't assign any numbers to keys in strings.json? ex) "1234"

Reply:
Ideally we don't do this as that makes the state unclear for people who are using templates to automate. Having a proper name also helps people who translate

Reply:
ODOR has deleted. I'll prepare Enum type and add next sensor PR. Thanks.